### PR TITLE
#941 - Add convex_hull of an array of polytopes

### LINKED ISSUE #941: convex_hull of polytopes
The documentation mentions that convex_hull is used to compute the convex hull of a polygon (2D only). However, convex_hull appears to run on a VPolytope of any dimensionality. The following quick check shows that adding 2 more dimensions to a vector of 2D points does not change the convex hull, suggesting that the function is not applicable at >2D.
```
V1 = VPolytope([rand(4) for i in 1:200])
V2 = VPolytope([v[1:2] for v in V1.vertices])

length(convex_hull(V1.vertices)) == length(convex_hull(V2.vertices)) #returns true
```
If this is indeed the case, then perhaps convex_hull should contain an assertion for dimensionality to prevent erroneous usage.

Also, given that convex_hull should not be used on higher dimensional polytopes, is there another way to calculate a convex hull? I am aware of the ConvexHull and ConvexHullArray types, but they seem to only work lazily. Is there a way to concretize a ConvexHull into an actual array of points?

Thanks!

### PR DESCRIPTION
**Pull request by SebastianGuadalupe**
Closes #941.

### CODE REVIEW

File: src/ConcreteOperations/convex_hull.jl [hunk 1]
```diff
@@ -476,3 +476,21 @@ function monotone_chain!(points::Vector{VN}; sort::Bool=true
...
+Compute the convex hull of an array of polytopes
+
+### Input
+
+- `U` -- UnionSetArray of polytopes
+
+### Output
+
+List of vectors containing the vertices of the convex hull of an array of
+the given polytopes.
```
Comment (schillic) [inline src/ConcreteOperations/convex_hull.jl:491]:
```suggestion
    convex_hull(U::UnionSetArray{N, <:AbstractPolytope{N}}) where {N}
    
Compute the convex hull of a union of a finite number of polytopes.

### Input

- `U` -- UnionSetArray of polytopes

### Output

A list of the vertices of the convex hull.
```

**Comment by mforets** [comment]
docs are green now... merging!

File: docs/src/lib/binary_functions.md [hunk 2]
```diff
@@ -53,6 +53,7 @@ convex_hull(::HPoly{N}, ::HPoly{N}) where {N<:Real}
 convex_hull(::VPolytope{N}, ::VPolytope{N}) where {N<:Real}
 convex_hull(::VPolygon{N}, ::VPolygon{N}) where {N<:Real}
 convex_hull(::Vector{VN}) where {N<:Real, VN<:AbstractVector{N}}
+convex_hull(U::UnionSetArray{N, PT}) where {N, PT<:AbstractPolytope{N}}
```
Comment (mforets) [inline docs/src/lib/binary_functions.md:56]:
```suggestion
convex_hull(::UnionSetArray{N, PT}) where {N, PT<:AbstractPolytope{N}}
```

Comment (mforets) [reply docs/src/lib/binary_functions.md:56]: @SebastianGuadalupe  note that the name of the variables are not needed here.

**Comment by mforets** [comment]
> I am not sure if removing the kwargs was good, but fine smiley

i agree; the kwargs can be used to pass a backend for  taking the convex hull, or additional options.

**Comment by mforets** [comment]
> i agree; the kwargs can be used to pass a backend for taking the convex hull, or additional options.

@SebastianGuadalupe  could you please try adding those again?

**Pull request merged**